tests/integration/components/bs-modal-test.js

```diff
@@ -331,4 +331,76 @@ module('Integration | Component | bs-modal', function (hooks) {
       assert.dom('.modal').doesNotExist();
     });
   });
+
+  test('Modal shows appropriate size respective to @size argument', async function (assert) {
+    this.set('size', null);
+    await render(hbs`
+      <BsModal @size={{this.size}} />
+    `);
+    assert.dom('.modal').exists({ count: 1 }, 'Modal exists.');
+    assert.dom('.modal-dialog').hasNoClass('modal-lg');
+    assert.dom('.modal-dialog').hasNoClass('modal-sm');
+    this.set('size', 'sm');
+    assert.dom('.modal-dialog').hasClass('modal-sm');
+    this.set('size', 'lg');
+    assert.dom('.modal-dialog').hasClass('modal-lg');
+  });
+
+  test('Modal can be centered vertically', async function (assert) {
+    this.set('position', 'top');
+    await render(hbs`
+      <BsModal @position={{this.position}} />
+    `);
+    assert.dom('.modal').exists({ count: 1 }, 'Modal exists.');
+    assert.dom('.modal-dialog').hasNoClass('modal-dialog-centered');
+    this.set('position', 'center');
+    assert.dom('.modal-dialog').hasClass('modal-dialog-centered');
+  });
+
+  test('shows scrollable modal with respective to @scrollable argument', async function (assert) {
+    this.set('scrollable', false);
+    await render(hbs`
+      <BsModal @scrollable={{this.scrollable}} />
+    `);
+    assert.dom('.modal').exists({ count: 1 }, 'Modal exists.');
+    assert.dom('.modal-dialog').hasNoClass('modal-dialog-scrollable');
+    this.set('scrollable', true);
+    assert.dom('.modal-dialog').hasClass('modal-dialog-scrollable');
+  });
+
+  test('Modal closes when escape key is pressed if @keyboard=true', async function (assert) {
+    await render(hbs`
+      <BsModal @keyboard={{true}} />
+    `);
+    assert.dom('.modal').exists({ count: 1 }, 'Modal exists.');
+    await triggerKeyEvent('.modal', 'keydown', 'Escape');
+    assert.dom('.modal').doesNotExist('Modal does not exist.');
+  });
+
+  test('Modal does not close when escape key is pressed if @keyboard=false', async function (assert) {
+    await render(hbs`
+      <BsModal @keyboard={{false}} />
+    `);
+    assert.dom('.modal').exists({ count: 1 }, 'Modal exists.');
+    await triggerKeyEvent('.modal', 'keydown', 'Escape');
+    assert.dom('.modal').exists({ count: 1 }, 'Modal exists.');
+  });
+
+  test('Modal closes when click is done outside the modal if @backdropClose=true', async function (assert) {
+    await render(hbs`
+      <BsModal @backdropClose={{true}} />
+    `);
+    assert.dom('.modal').exists({ count: 1 }, 'Modal exists.');
+    await click('*');
```

Comment:
The selector `*` should trigger a click on all events. Or just on the first one, which is matched by the selector? I'm not sure. But I think `this.element` is more precise in any case.

```suggestion
    await click(this.element);
```

Reply:
`this.element` does not work, I have just tried it. `await click("*")` will trigger click event only on the first element which is matched by the selector.

We have already used this approach when closing the Bootstrap dropdown:
https://github.com/ember-bootstrap/ember-bootstrap/blob/master/tests/integration/components/bs-dropdown-test.js#L110